tailnet/test/integration/integration.go

```diff
@@ -111,13 +111,43 @@ type SimpleServerOptions struct {
 
 var _ ServerStarter = SimpleServerOptions{}
 
+type connManager struct {
+	sync.Mutex
+	conns map[uuid.UUID]net.Conn
+}
+
+func (c *connManager) Add(id uuid.UUID, conn net.Conn) func() {
+	c.Lock()
+	c.conns[id] = conn
+	c.Unlock()
+	return func() {
+		c.Lock()
+		delete(c.conns, id)
+		c.Unlock()
+	}
+}
+
+func (c *connManager) Clear() {
```

Comment:
Probably should be called `CloseAll` or something to accurately reflect what it does